lib/jekyll/commands/serve/servlet.rb

```diff
@@ -21,10 +21,36 @@ def initialize(server, root, callbacks)
         # up with a different preference on which comes first.
 
         def search_file(req, res, basename)
-          # /file.* > /file/index.html > /file.html
+          # /file.* -> /file.html
```

Comment:
I don't think the comment *meant* changing `/file.*` into `/file.index.html` and then into `/file.html` either.
Its *probably* documenting the order of traversal.